src/rendergraph/scenegraph/backend/baseattributeset.cpp

```diff
@@ -9,6 +11,8 @@ int toQSGGeometryType(const PrimitiveType& t) {
         return QSGGeometry::FloatType;
     case PrimitiveType::UInt:
         return QSGGeometry::UnsignedIntType;
+    default:
+        throw std::runtime_error("not implemented");
```

Comment:
two nits: 
1. exceptions in Mixxx are a touchy subject. 
2. this essentially silences `-Wswitch` which is usually undesirable. If you need to add a return value or throw (because the compiler complains otherwise), just do it after the switch/outside.

Reply:
Yeah, I'm definitely keen to control the proliferation of pesky exception. I've added this one to align with the code style of the rendergraph (see [here](https://github.com/mixxxdj/mixxx/blob/89bf0a23bfa795cb4b002f5443a8079ea0797de4/src/rendergraph/opengl/backend/basegeometrynode.cpp#L21), [here](https://github.com/mixxxdj/mixxx/blob/89bf0a23bfa795cb4b002f5443a8079ea0797de4/src/rendergraph/scenegraph/geometry.cpp#L17) or [here](https://github.com/mixxxdj/mixxx/blob/89bf0a23bfa795cb4b002f5443a8079ea0797de4/src/rendergraph/scenegraph/geometry.cpp#L28)). I'm almost tempted to replace these with a `VERIFY_OR_DEBUG_ASSERT` but felt like potential cleanup task to keep out of this PR scope.

Happy to "break" the rendergraph style already here tho if you think this is sensible

Reply:
Oh, I'm sorry. I had lost context and thought we were switching on our our own sg `enum`. The discussion is void then.